sklearn/tests/test_naive_bayes.py

```diff
@@ -945,3 +952,678 @@ def test_n_features_deprecation(Estimator):
 
     with pytest.warns(FutureWarning, match="`n_features_` was deprecated"):
         est.n_features_
+
+
+def test_cwnb_union():
```

Comment:
i would split this test in more atomic tests. Basically split each condition you test here in dedicated tests

Reply:
I would also test this in the case you set priors manually in ColumnwiseNB

Reply:
> i would split this test in more atomic tests. Basically split each condition you test here in dedicated tests

Done.

> I would also test this in the case you set priors manually in ColumnwiseNB

Something strange is going on, which I am looking into more closely. I will investigate and report on this soon.

Reply:
> I would also test this in the case you set priors manually in ColumnwiseNB

Done. While adding this test, I noted that it is easy to forget to specify priors in each subestimator. I further discuss it [in the message below](https://github.com/scikit-learn/scikit-learn/pull/22574#issuecomment-1146756994).